### PR TITLE
Failure cluster issue assignment fixes

### PR DESCRIPTION
**Pull request by cjwagner**
Fixes to issue assignment behavior:
-OwnerList now ignores leading and trailing spaces for fields in test_owners.csv.
-OwnerList no longer assigns to the default owners if it can't find owners for a specific test.

Also added more logging for issue creation (Full issue contents is now logged).

/assign @rmmh

### CODE REVIEW

File: mungegithub/mungers/issue-creator.go [hunk 1]
```diff
@@ -292,10 +292,15 @@ func (c *IssueCreator) Sync(issue Issue) {
 	}
 	labels = c.filterValidLabels(labels)
 
+	var dryRunStr string
+	if c.config.isDryRun() {
+		dryRunStr = "--DryRun--"
```
Comment (rmmh) [inline mungegithub/mungers/issue-creator.go]: this isn't necessary. mutations printed during dry runs are obviously fake

**Comment by rmmh** [comment]
Do you have a corresponding PR in k8s to remove the DEFAULT line?

**Comment by cjwagner** [comment]
@k8s-bot bazel test this

File: mungegithub/mungers/issue-creator.go (hunk 1 above)
Comment (cjwagner) [inline mungegithub/mungers/issue-creator.go]: done

**Comment by cjwagner** [comment]
@rmmh I haven't made a corresponding PR to remove the default entry from test_owners.csv yet because I saw that test_owners.csv was used by some other files in k8s, but it looks like its just the python script to update and validate the csv file so its fine to change that. I'll make that PR.

**Pull request merged**